### PR TITLE
New calibration integration test

### LINKED ISSUE #384: start writing integration tests for the new code path
https://github.com/MESMER-group/mesmer/blob/7667fe92a545111adf919cfd405f1a380dfc086b/tests/integration/test_calibrate_mesmer.py#L74

does integration tests for the legacy code path. We should write the same tests for the new code path. (NOTE: I have notebooks doing this - we can start from there and integrate them into MESMER proper).

### PR DESCRIPTION
**Pull request by veni-vidi-vici-dormivi**
Closes #384

### CODE REVIEW

**Comment by veni-vidi-vici-dormivi** [comment]
This now only covers the very basic test case of one ensemble member, one scenario. But we can build on it.

File: tests/integration/test_calibrate_mesmer_newcodepath.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+        PARAMS_PATH = TEST_PATH / "params"
+
+        fN_bundle = TEST_PATH / "test-mesmer-bundle.pkl"
+
+        bundle = joblib.load(fN_bundle)
+
+        def load_params(*folders, file):
+            fN = os.path.join(PARAMS_PATH, *folders, file)
+
+            return joblib.load(fN)
```
Comment (mathause) [inline tests/integration/test_calibrate_mesmer_newcodepath.py:117]: Only the bundle was commited - can you try to get rid of the `load_params` part? I think most what I read from the "params" is actually in the `bundle`.

See example below. Probably some are missing, we need to figure out what to do about them once we know which (either commit them as well or don't test).

File: tests/integration/test_calibrate_mesmer_newcodepath.py [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+    HIST_PERIOD = slice("1850", "2014")
+    PROJ_PERIOD = slice("2015", "2100")
+
+    LOCALISATION_RADII = range(1750, 2001, 250)
+
+    test_esms = ["IPSL-CM6A-LR"]
+    test_cmip_generation = 6
+
+    for esm in test_esms:
```
Comment (mathause) [inline tests/integration/test_calibrate_mesmer_newcodepath.py:102]: Let's remove this loop for now.

File: tests/integration/test_calibrate_mesmer_newcodepath.py [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+        )
+
+        # TODO: better seperate testing from calculating maybe
+        params_gv_T = load_params(
+            "global",
+            "global_variability",
+            file=f"params_gv_AR_tas_{esm}_hist_{scenario}.pkl",
+        )
+
+        np.testing.assert_allclose(params_gv_T["AR_int"], global_ar_params.intercept)
```
Comment (mathause) [inline tests/integration/test_calibrate_mesmer_newcodepath.py:228]:
```suggestion
        np.testing.assert_allclose(bundle["params_gv"]["AR_int"], global_ar_params.intercept)
```

e.g. this one.

**Review by mathause — COMMENTED** [review]
Good start. I agree it would be good to start separating the calibration from the tests. Maybe by first moving the stuff to the end?

**Review by mathause — COMMENTED** [review]
This is coming along nicely!

**Review by mathause — APPROVED** [review]
I think this is ready! Good job.

**Comment by veni-vidi-vici-dormivi** [comment]
Thank you 😊

**Pull request merged**